core/src/com/unciv/logic/map/MapGenerator.kt

```diff
@@ -519,7 +524,7 @@ class MapGenerator {
         if (suitableLocations.isNotEmpty()) {
             val location = suitableLocations.random()
             location.naturalWonder = wonder.name
-            location.baseTerrain = Constants.plains
+            location.baseTerrain = wonder.turnsInto!!
             location.terrainFeature = null
```

Comment:
These look like 5 lines that occur over and over for each nat wonder, could probably be unified into a single function...?
like trySpawn(wonder:Terrain, suitableLocations:List<TileInfo>):TileInfo?
And then for special wonders that affect surrounding tiles you can use the return value to determine if it spawned or not and what the location is